ui/src/authentication/services/AuthenticationService.js

```diff
@@ -1,30 +1,77 @@
+import { http } from '@api';
+import { sessionManager } from './SessionManager';
+
 class AuthenticationService {
-  getTokenFromDOM = () => {
-    const element = document.getElementsByName('authorized_token');
-    if (element.length > 0 && element[0].hasAttribute('value')) {
-      return element[0].value;
+  loginWithOauthProvider = (nextUrl, providerUrl) => {
+    localStorage.setItem('ILS_USER_WAS_LOGGEDIN', true);
```

Comment:
This was a long discussion... I am not a fan of this kind of states, but I guess it is needed.

Basically, if the user was, at some point in time, logged in, you have 2 possible scenarios:
1. he has a valid session
2. ha has not a valid session
but we don't know which one. You could call a random authenticated endpoint to test this and validate the session, but it makes probably sense to call the `/me`.

If case 1, then you can fetch a fresh version of the user profile. If case 2, then it sets the session to anonymous.
Unless you see any corner case, it should not lead to inconsistencies...

Reply:
This is just to avoid the extra `/me` in a best effort strategy. But if a 401 is being thrown somewhere in the application the user will be forced to login again so it will be safe :)